backends/vllm/src/model_executor.rs

```diff
@@ -394,10 +562,32 @@ impl ModelThreadDispatcher {
         trace!("Sending new `ExecuteModelRequest` to model executor task");
 
         let (sender, receiver) = oneshot::channel();
-        let command = ModelThreadCommand { request, sender };
 
-        if let Err(e) = self.sender.send(command) {
-            error!("Could not send command to model core, it might be shutting down: {e}");
+        #[cfg(not(feature = "nccl"))]
+        {
+            let command = ModelThreadCommand { request, sender };
+            if let Err(e) = self.sender.send(command) {
+                error!("Could not send command to model core, it might be shutting down: {e}");
+            }
+        }
+        #[cfg(feature = "nccl")]
```

Comment:
I have an idea, too much code is being split into nccl and non nccl, but most of the code can be made similar, for non nccl you can also have vector of sender (it just will be 1 sender) and you can use the same code then
I think it's worth for the readability of the code.

Reply:
Definitely, appreciate it !